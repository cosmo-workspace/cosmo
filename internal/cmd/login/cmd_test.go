package login

import (
	"bytes"
	"testing"

	"github.com/cosmo-workspace/cosmo/pkg/cli"
	. "github.com/cosmo-workspace/cosmo/pkg/snap"
	. "github.com/onsi/ginkgo/v2"
	. "github.com/onsi/gomega"

	"github.com/spf13/cobra"
)

func TestCommandLogin(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "cosmoctl login suite")
}

var _ = Describe("help", func() {
	It("should match snapshot", func() {
		cmd := &cobra.Command{}
		out := bytes.Buffer{}
		cmd.SetOut(&out)
		AddCommand(cmd, cli.NewRootOptions())
		cmd.SetArgs([]string{"login", "--help"})
		err := cmd.Execute()
		Expect(err).ShouldNot(HaveOccurred())
		Expect(out.String()).To(MatchSnapShot())
	})
})
