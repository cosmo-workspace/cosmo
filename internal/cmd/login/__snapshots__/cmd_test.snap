['help should match snapshot 1']
SnapShot = """
Login to COSMO Dashboard Server

Usage:
   login USER_NAME [flags]

Examples:

  # interactive mode
  cosmoctl login

  # non interactive mode
  echo $PASSWORD | cosmoctl login USER_NAME --dashboard-url https://DASHBOARD_URL --password-stdin


Flags:
      --again            login again
  -h, --help             help for login
      --password-stdin   input new password from stdin pipe
"""
